src/theme/components/mailingListSubscription/Form.tsx

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright 2021 Bonitasoft S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import styled from 'styled-components';
+import colors from '../../colors.json';
+import React, { BaseSyntheticEvent, useState } from 'react';
+import addToMailchimp, { MailchimpResponse } from 'gatsby-plugin-mailchimp';
+import { ContainerContentProps } from './ContainerContent';
+import { Button } from 'rebass/styled-components';
+
+export interface FormProps extends ContainerContentProps {
+  onSubmit: (response: MailchimpResponse, submitted: boolean) => void;
+}
+
+interface InputProps extends ContainerContentProps {
+  error?: boolean;
+}
+
+export const Form = (props: FormProps): JSX.Element => {
+  const [error, setError] = useState(false);
+  const [email, setEmail] = useState('');
+
+  const handleSubmit = async (e: BaseSyntheticEvent): Promise<void> => {
+    e.preventDefault();
+
+    if (!isEmailValid()) {
+      setError(true);
+      return;
+    }
+
+    const response = await addToMailchimp(email);
+
+    const failed = response.result === 'error';
+    setError(failed);
+    props.onSubmit(response, !failed);
+  };
+
+  const isEmailValid = (): boolean => {
```

Comment:
❓ why not using email validation provided by an email input?

Reply:
I just copy the example I use. I didn't see that we can add a pattern on the input tag.
I'll try it 😉

Reply:
If we use it, we can't set the error int the Form state, and we can't have the animation on the field.
To avoid to change too many thing, I propose to test it on another PR 🙂

Reply:
Ok, let's do this later or not